src/core/ext/filters/client_channel/client_channel.c

```diff
@@ -774,6 +735,8 @@ static void cc_destroy_channel_elem(grpc_exec_ctx *exec_ctx,
 
 #define CANCELLED_CALL ((grpc_subchannel_call *)1)
 
+#define MAX_WAITING_BATCHES 6
```

Comment:
where's that magic 6 coming from? Why 6 and not 10? We should at the very least have a comment. Ideally we'd get that max number from some other file where it's uniquely defined.